res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1752,22 +1752,31 @@ PioneerDDJSX.playLed = function(value, group, control) {
 };
 
 PioneerDDJSX.wheelLeds = function(value, group, control) {
+    // Timing calculation is handled in seconds!
     var deck = PioneerDDJSX.channelGroups[group],
         duration = engine.getValue(group, "duration"),
-        remainingTime = duration - (value * duration),
-        speed = 0x28,
-        wheelPos = 0x01;
+        elapsedTime = value * duration,
+        remainingTime = duration - elapsedTime,
+        refsPerSecond = PioneerDDJSX.scratchSettings.vinylSpeed / 60,
+        speed = parseInt(refsPerSecond * 0x48),
+        wheelPos = 0x00;
 
+    // wheelPos = 0x48 : all LEDs are lit, wheelPos = 0x00 : all LEDs off
```

Comment:
This is better. It could still be improved by assigning a variable to `0x48` named something like `numLightsInCircle` (you might come up with a better name) and use that in the calculations.